nifi-nar-bundles/nifi-standard-bundle/nifi-standard-processors/src/main/java/org/apache/nifi/processors/standard/GrokParser.java

```diff
@@ -0,0 +1,243 @@
+    private static final String APPLICATION_JSON = "application/json";
+
+    public static final PropertyDescriptor GROK_EXPRESSION = new PropertyDescriptor
+            .Builder().name("Grok Expression")
+            .description("Grok expression")
+            .required(true)
+            .addValidator(StandardValidators.NON_EMPTY_VALIDATOR)
+            .build();
+
+    public static final PropertyDescriptor GROK_PATTERN_FILE = new PropertyDescriptor
+            .Builder().name("Grok Pattern file")
+            .description("Grok Pattern file definition")
+            .required(false)
+            .addValidator(StandardValidators.FILE_EXISTS_VALIDATOR)
+            .build();
+
+    public static final PropertyDescriptor DESTINATION = new PropertyDescriptor.Builder()
+            .name("Destination")
+            .description("Control if Grok output value is written as a new flowfile attribute  " +
+                    "or written in the flowfile content. Writing to flowfile content will overwrite any " +
+                    "existing flowfile content.")
+            .required(true)
+            .allowableValues(DESTINATION_ATTRIBUTE, DESTINATION_CONTENT)
+            .defaultValue(DESTINATION_ATTRIBUTE)
+            .build();
+
+    public static final PropertyDescriptor CHARACTER_SET = new PropertyDescriptor
+            .Builder().name("Character Set")
+            .description("The Character Set in which the file is encoded")
+            .required(true)
+            .addValidator(StandardValidators.CHARACTER_SET_VALIDATOR)
+            .defaultValue("UTF-8")
+            .build();
+
+    public static final PropertyDescriptor MAX_BUFFER_SIZE = new PropertyDescriptor
+            .Builder().name("Maximum Buffer Size")
+            .description("Specifies the maximum amount of data to buffer (per file) in order to apply the Grok expressions. Files larger than the specified maximum will not be fully evaluated.")
+            .required(true)
+            .addValidator(StandardValidators.DATA_SIZE_VALIDATOR)
+            .addValidator(StandardValidators.createDataSizeBoundsValidator(0, Integer.MAX_VALUE))
+            .defaultValue("1 MB")
+            .build();
+
+    public static final Relationship REL_MATCH = new Relationship.Builder()
+            .name("matched")
+            .description("FlowFiles are routed to this relationship when the Grok Expression is successfully evaluated and the FlowFile is modified as a result")
+            .build();
+
+    public static final Relationship REL_NO_MATCH = new Relationship.Builder()
+            .name("unmatched")
+            .description("FlowFiles are routed to this relationship when no provided Grok Expression matches the content of the FlowFile")
+            .build();
+
+    private List<PropertyDescriptor> descriptors;
+
+    private Set<Relationship> relationships;
+
+    private static final ObjectMapper objectMapper = new ObjectMapper();
+
+    private Grok grok;
```

Comment:
These member variables are not thread-safe. They should be defined inline where needed, or properly protected via Java's concurrency mechanisms.